src/checkpoint/checkpoint.ts

```diff
@@ -151,11 +231,33 @@ export default class Checkpoint {
           }
         }
       }
+
+      if (foundContractData) {
+        // no need to await checkpoints inserts, so insert async and log if there is an error
```

Comment:
I would recommend to await here, if you miss a checkpoint at block 1000 because you restarted the server before this function get resolved but latest sync block was updated and then you find a checkpoint at block 1010. If this happen the block 1000 will be skipped and events will be missing in the replay.

Reply:
That is a valid point, I'll await it. Fixed here: https://github.com/snapshot-labs/sx-api/pull/39/commits/630131f2d76d63f2ea61ae23819c50ce3d470e27